### PR TITLE
save manifest to file

### PR DESCRIPTION
**Pull request by quinnwai**
Addresses
- #83 to save a copy of the manifest when running `vrs_bulk annotate` so that there is a history of what configuration / files  were used for that run
- #82 to couple the logs, manifest, and metrics files by timestamp
- #71 (by doing the above)

Once we do this work, we have associated create CLI to make CAF objects given a timestamp, which makes for less manual work for a downstream analyst.

### CODE REVIEW

File: vrs_anvil/annotator.py [hunk 1]
```diff
@@ -171,13 +170,14 @@ def annotate_all(
...
                     PARAMETERS: {"fmt": result.fmt, "var": result.var},
-                    VRS_OBJECT: allele.model_dump(exclude_none=True)
+                    VRS_OBJECT: allele.model_dump(exclude_none=True),
                 }
 
                 metrics[file_path][METAKB_HITS] += 1
 
     _logger.info("annotate_all: Finished processing results.")
 
+    metrics[TOTAL]["timestamp_str"] = timestamp_str
```
Comment (bwalsh) [inline vrs_anvil/annotator.py]: make timestamp_str a constant

**Review by bwalsh — CHANGES_REQUESTED** [review]
Code looks great.
* needs some tests 
* update readme.
* do we need a short user guide?

**Comment by quinnwai** [comment]
@bwalsh responding to points...

> * needs some tests

Sounds good. Will work on adding tests for suffix usage, as well as adding tests that didn't exist before including `--scatter` tests and `vrs_bulk ps` tests. 

> * update readme.

Readme is updated a little bit in #87, if you want I could just target 87 to merge into here and add more? Let me know

> * do we need a short user guide?

Similar to above, I can add a commit once that's done so there's no conflicts on the readme?

File: vrs_anvil/annotator.py (hunk 1 above)
Comment (quinnwai) [reply vrs_anvil/annotator.py]: updated

**Comment by quinnwai** [comment]
@bwalsh Updated it to add tests for `ps`, `annotate --scatter`, and doing refactoring of tests to include stubbing out functions, CLI tests, working in a tmp directory.

Ready for your re-review

**Comment by bwalsh** [comment]
```$ pytest tests/unit/
================================================================================================ test session starts =================================================================================================
platform darwin -- Python 3.12.1, pytest-8.2.1, pluggy-1.5.0
rootdir: /Users/walsbr/anvil/vrs_anvil_toolkit
configfile: pytest.ini
collected 17 items

tests/unit/test_1000g_vcf.py .                                                                                                                                                                                 [  5%]
tests/unit/test_annotator.py ......                                                                                                                                                                            [ 41%]
tests/unit/test_cli.py F....                                                                                                                                                                                   [ 70%]
tests/unit/test_coding_conventions.py .                                                                                                                                                                        [ 76%]
tests/unit/test_collect_vcfs.py .                                                                                                                                                                              [ 82%]
tests/unit/test_metakb.py .                                                                                                                                                                                    [ 88%]
tests/unit/test_translator.py .                                                                                                                                                                                [ 94%]
tests/unit/test_vcf_to_gnomad.py .                                                                                                                                                                             [100%]

====================================================================================================== FAILURES ======================================================================================================
__________________________________________________________________________________________________ test_cli_version __________________________________________________________________________________________________

    def test_cli_version():
        """Test that version can be called and printed"""
        runner = CliRunner()
        result = runner.invoke(cli, "--version")
        expected_strings = ["version"]
        print(result.output)
        for expected_string in expected_strings:
>           assert (
                expected_string in result.output
            ), f"Should have printed {expected_string}"
E           AssertionError: Should have printed version
E           assert 'version' in ''
E            +  where '' = <Result RuntimeError("'vrs_anvil' is not installed. Try passing 'package_name' instead.")>.output

tests/unit/test_cli.py:80: AssertionError
------------------------------------------------------------------------------------------------ Captured stdout call ------------------------------------------------------------------------------------------------

============================================================================================== short test summary info ===============================================================================================
FAILED tests/unit/test_cli.py::test_cli_version - AssertionError: Should have printed version```

was package renamed to `vrs_bulk`

**Comment by quinnwai** [comment]
@bwalsh fixed failing --version test and bumped version. Ready for your re-review

**Review by bwalsh — APPROVED** [review]

**Pull request merged**